extension/src/commands/CreateProjectCommand.ts

```diff
@@ -48,12 +49,38 @@ export class CreateProjectCommand extends Command {
             const success = await this.runSteps(metadata);
             if (success) {
                 await this.createProject(metadata);
-                const openInNewWindow = folders !== undefined;
-                vscode.commands.executeCommand(
-                    "vscode.openFolder",
-                    vscode.Uri.file(metadata.targetFolder),
-                    openInNewWindow
-                );
+                const hasOpenFolder = folders !== undefined;
+                let openProjectBehaviour = getProjectOpenBehaviour();
+                if (openProjectBehaviour === ProjectOpenBehaviourValue.INTERACTIVE) {
+                    const insideWorkspace: boolean | undefined =
+                        folders?.find((workspaceFolder) =>
+                            metadata.targetFolder.startsWith(workspaceFolder.uri?.fsPath)
+                        ) !== undefined;
+                    const candidates: string[] = [
+                        ProjectOpenBehaviourValue.OPEN,
+                        hasOpenFolder && insideWorkspace === false
+                            ? ProjectOpenBehaviourValue.ADDTOWORKSPACE
+                            : undefined,
+                    ].filter(Boolean) as string[];
+                    const choice = await vscode.window.showInformationMessage(
+                        `Gradle project [${metadata.projectName}] is created under: ${metadata.targetFolder}`,
+                        ...candidates
+                    );
+                    if (choice) {
+                        openProjectBehaviour = choice;
+                    }
+                }
+                if (openProjectBehaviour === ProjectOpenBehaviourValue.OPEN) {
+                    vscode.commands.executeCommand(
+                        "vscode.openFolder",
+                        vscode.Uri.file(metadata.targetFolder),
+                        hasOpenFolder
+                    );
+                } else if (openProjectBehaviour === ProjectOpenBehaviourValue.ADDTOWORKSPACE) {
+                    vscode.workspace.updateWorkspaceFolders(folders!.length, null, {
```

Comment:
what if here `folders` is undefined. Say the setting is pointed to "ADDTOWORKSPACE", and you create a project without opening any folder.

Reply:
See: https://github.com/microsoft/vscode-gradle/pull/1282/files#diff-99dbae36046fe7b4147fc646659041025bc994a4b7762f5e08713881e87e0cebR61 we will not show this option when `folders` is undefined.

Reply:
https://github.com/microsoft/vscode-gradle/pull/1282/files#diff-99dbae36046fe7b4147fc646659041025bc994a4b7762f5e08713881e87e0cebR54
You won't enter that block if the setting is explicitly set to `ADDTOWORKSPACE`

Reply:
you're right, let me correct it.